src/overlay/Peer.cpp

```diff
@@ -1141,6 +1175,21 @@ Peer::recvTxSet(StellarMessage const& msg)
     ZoneScoped;
     auto frame = TxSetFrame::makeFromWire(mApp, msg.txSet());
     mApp.getHerder().recvTxSet(frame->getContentsHash(), frame);
+    CLOG_INFO(Overlay,
+              "Peer::recvTxSet received {}, trying to fulfill pending requests",
+              hexAbbrev(frame->getContentsHash()));
+    auto pendingTxSetRequests =
+        mApp.getOverlayManager().getPendingTxSetRequests();
+
+    for (auto& weakPeer : pendingTxSetRequests[frame->getContentsHash()])
```

Comment:
I'm not 100% percent sure, but I think this unchecked map access may open up a DOS angle. Probably want to check with @marta-lokhova, but it looks like this assumes the incoming message asks for a pending TxSet hash that we currently have in `pendingTxSetRequests`. I don't think this assumption is valid given byzantine nodes or dropped messages from a well intentioned node. You should check that the hash exists in the map and not do an unchecked access.